components/Button/Button.js

```diff
@@ -0,0 +1,63 @@
+function createButton(options = {}) {
+  const {
+    icon = null,
+    label = null,
+    size = "medium", // 'small', 'medium', 'large'
+    onClick = null,
+  } = options;
+
+  const button = document.createElement("button");
+  button.className = `button button--${size}`;
+
+  // 아이콘이 있는 경우 아이콘 추가
+  if (icon) {
+    const iconElement = document.createElement("img");
+    iconElement.className = `button__icon icon--${size}`;
+    iconElement.src = icon;
+    iconElement.alt = "button-icon";
+    button.appendChild(iconElement);
+  }
+
+  // 라벨이 있는 경우 텍스트 추가
+  if (label) {
+    const labelElement = document.createElement("span");
+
+    // 사이즈별 폰트 클래스 적용
+    let fontClass = "";
+    switch (size) {
+      case "large":
+        fontClass = "font-serif-48";
+        break;
+      case "medium":
+        fontClass = "font-semibold-16";
+        break;
+      case "small":
+        fontClass = "font-semibold-12";
+        break;
+      default:
+        fontClass = "font-semibold-16";
+    }
```

Comment:
switch문 대신 size에 따른 fontClass를 상수화해서 접근해도 괜찮을 것 같아요
```suggestion
    const 폰트사이즈 = {
        large: "font-serif-48",
        medium: "font=semibold-16",
        small: "fontsemibold-12",
        };
    fontClass = 폰트사이즈[size];
```

Reply:
좋은 방식인것 같습니다! 감사합니다